### PR TITLE
1006 single file install

### PR DESCRIPTION
**Pull request by tuxpiper**
Fixes ushahidi/platform#1006

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/222)

### CODE REVIEW

File: docker/build.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+RUN git config --global url."https://github.com/".insteadOf git@github.com:
+
+RUN npm install -g gulp@3.9.0 gulp-notify@2.2.0 grunt-cli@0.1.13
+
+RUN mkdir -p /var/app
+WORKDIR /var/app
+COPY package.json /var/app
+RUN npm install
```
Comment (rjmackay) [inline docker/build.Dockerfile]: Why do you copy package.json here and run `npm install` but then run `npm install` etc in build.run.sh too?

File: docker/build.run.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  fi
+}
+check_vols_out() {
+  if [ ! -d /vols/out ]; then
+    echo "No /vols/out for output!"
+    exit 1
+  fi
+}
+
+function sync {
```
Comment (rjmackay) [inline docker/build.run.sh]: Can you add some comments re: whats going on here?

File: docker/deploy.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+# ==> Add wrapper script
+COPY ./docker/docker_run.sh /docker_run.sh
+
+# ==> Turn off host key checking for Ansible
+ENV ANSIBLE_HOST_KEY_CHECKING False
+
+# ==> Add deploy key to container
+RUN mkdir -m 700 -p "$HOME/.ssh"
+
+ENTRYPOINT [ "/bin/bash", "/docker_run.sh" ]
```
Comment (rjmackay) [inline docker/deploy.Dockerfile]: rename to deploy.run.sh to avoid confusion?

File: docker/build.run.sh [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    echo "- tmp"
+  } > /tmp/rsync_exclude
+  rsync -ar --exclude-from=/tmp/rsync_exclude --delete-during /vols/src/ ./
+}
+
+build() {
+  npm install
+  gulp transifex-download
+  gulp build
+  cp ./server/rewrite.htaccess ./server/www/
```
Comment (rjmackay) [inline docker/build.run.sh]: FYI there is a `gulp release` command we used before to build a package. I'm not sure if thats useful here or if we should drop it now?
Run like `gulp release --version-suffix=${TRAVIS_TAG:-$TRAVIS_COMMIT}` to append the tag or commitid to the built zip

File: docker/build.run.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+bundle() {
+  check_vols_out
+  local version=${GITHUB_VERSION:-${CI_BRANCH:-v0.0.0}}
+  mkdir /tmp/ushahidi-platform-client-bundle-${version}; rsync -ar ./server/www/ /tmp/ushahidi-platform-client-bundle-${version}/
```
Comment (rjmackay) [inline docker/build.run.sh]: FYI there is a `gulp release` command we used before to build a package. I'm not sure if thats useful here or if we should drop it now?
Run like `gulp release --version-suffix=${TRAVIS_TAG:-$TRAVIS_COMMIT}` to append the tag or commitid to the built zip

**Comment by rjmackay** [comment]
@tuxpiper this looks good. A few comments on the code itself. Do you have time to resolve the conflicts and land? or shall I grab this tomorrow?

File: docker/build.Dockerfile (hunk 1 above)
Comment (tuxpiper) [inline docker/build.Dockerfile]: In this specific usecase it's redundant (I actually repurposed this code from a dev environment that I set up with docker).

The idea is that, each time the container is (re-)started, files in the container are synced with the ones in the dev working directory, npm install is run to cope with any changes in dependencies.

**Comment by tuxpiper** [comment]
Ok, there's some stuff to do to handle the comments and the merge conflicts, will get to it asap

**Comment by tuxpiper** [comment]
@rjmackay addressed the comments and resolved conflicts

**Comment by rjmackay** [comment]
:lgtm:

---

Reviewed 2 of 8 files at r2, 5 of 8 files at r3, 4 of 8 files at r4, 2 of 7 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/222#-:-KLP96jy2mqF7i2Y35sA:1417088721)*

**Pull request merged**